tools/sharkbait/src/sharkbait/permissions.clj

```diff
@@ -21,9 +22,10 @@
   "Sets the allowed actions for a permission definition."
   [permission-def]
   (let [delegate (.getAttributeDefActionDelegate permission-def)]
-    (.configureActionList delegate ["read" "write" "own"])
-    (add-implied-action delegate "own" "write")
-    (add-implied-action delegate "write" "read"))
+    (.configureActionList delegate [read write own originator])
+    (add-implied-action delegate originator own)
```

Comment:
Yeah, I guess I could have explained the idea a little more clearly. I'm trying to enable the population of the `Shared with Me` category in the DE. This category will contain all apps that have been shared with the user either directly or (eventually) via user-managed groups. The problem is that when `read`, `write` and `own` are the only permission levels, there's no way to distinguish apps that the user has integrated from apps to which the user has been given `own` permissions by other users.

There are other options for ensuring that apps the user has integrated don't appear in the `Shared with Me` category, of course. We could:

1. Filter out apps that are in the user's `Apps Under Development` category.
2. Add a real `Shared with Me` category in the DE database.
3. Add a way to reliably identify the original integrator of an app in the DE database.
4. Add an `app-integrators` role to grouper that is reserved for permission assignments given to the original integrator of an app.

There are probably other options, but those are the ones that immediately come to mind. The problem with the first and second alternatives is that they will no longer work when the current app category system goes away. The third alternative would work, but the DE database seems like the wrong place to put this information since we're placing all other information about app sharing in Grouper. The last alternative is something that I hadn't considered before I added the new action. It might be a reasonable alternative, but I haven't been able to identify any advantage that this method has over the new action name.

I should mention that this action name is intended to be completely hidden from users. Its only purpose is to help the apps service determine which apps should go in the Shared with Me virtual category.

I'm not particularly attached to the `originator` name. It just seemed general enough to mean either `integrator` or `submitter`. The term, `lead`, doesn't really help me to identify the purpose of the action name either. I'm open to suggestions on this.

The original intent of the new action name wasn't to prevent other users from revoking permissions from the original integrator, but that is another potential use of it.